internal/controller/pkg/revision/establisher.go

```diff
@@ -142,7 +143,9 @@ func (e *APIEstablisher) validate(ctx context.Context, objs []runtime.Object, pa
 	if parent.GetWebhookTLSSecretName() != nil {
 		s := &corev1.Secret{}
 		nn := types.NamespacedName{Name: *parent.GetWebhookTLSSecretName(), Namespace: e.namespace}
-		if err := e.client.Get(ctx, nn, s); err != nil {
+		// Use uncached client because there might be many secrets in a cluster.
```

Comment:
Is the motivation only being "many", or also "sensitive"? 
Wondering if we would need the same for configmaps as well.

Reply:
Just many. Think for 25000 secrets in some big cluster.